File: datasets/CJBench/Java/DeepseekCommented/CircularBuffer.java
import java.util.Scanner;
public class Main {
    private final Item[] buffer;
    private final CircularPointer putPointer;
    private final CircularPointer getPointer;
    private int size;
    public Main(int size) {
        if (size <= 0) {
            throw new IllegalArgumentException("Buffer size must be positive");
        }
        this.buffer = (Item[]) new Object[size];
        this.putPointer = new CircularPointer(0, size);
        this.getPointer = new CircularPointer(0, size);
        this.size = 0;
    }
    // Checks if the buffer is empty
    public boolean isEmpty() {
        return size == 0;
    }
    // Checks if the buffer is full
    public boolean isFull() {
        return size == buffer.length;
    }
    // Retrieves an item from the buffer
    public Item get() {
        if (isEmpty()) {
            return null;
        }
        Item item = buffer[getPointer.getAndIncrement()];
        size--;
        return item;
    }
    // Adds an item to the buffer
    public boolean put(Item item) {
        if (item == null) {
            throw new IllegalArgumentException("Null items are not allowed");
        }
        if (isFull()) {
            getPointer.getAndIncrement(); 
        } else {
            size++;
        }
        buffer[putPointer.getAndIncrement()] = item;
        return true;
    }
    private static class CircularPointer {
        private int pointer;
        private final int max;
        CircularPointer(int pointer, int max) {
            this.pointer = pointer;
            this.max = max;
        }
        // Increments the pointer and returns the old value
        public int getAndIncrement() {
            int tmp = pointer;
            pointer = (pointer + 1) % max;
            return tmp;
        }
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int bufferSize = scanner.nextInt();
        int numberOfOperations = scanner.nextInt();
        Main<String> buffer = new Main<>(bufferSize);
        for (int i = 0; i < numberOfOperations; i++) {
            String operation = scanner.next();
            if (operation.equals("PUT")) {
                String element = scanner.next();
                buffer.put(element);
                System.out.println("Added: " + element);
            } else if (operation.equals("GET")) {
                String result = buffer.get();
                System.out.println("Retrieved: " + (result != null ? result : "null"));
            } else if (operation.equals("ISEMPTY")) {
                System.out.println("Is Empty: " + buffer.isEmpty());
            } else if (operation.equals("ISFULL")) {
                System.out.println("Is Full: " + buffer.isFull());
            }
        }
        scanner.close();
    }
}

File: translations/CJBench/J2P/Uncommented/starcoder/Code/CircularBuffer.py

class Main:
    def __init__(self, size):
        if size <= 0:
            raise ValueError("Buffer size must be positive")
        self.buffer = [None] * size
        self.putPointer = CircularPointer(0, size)
        self.getPointer = CircularPointer(0, size)
        self.size = 0

    def isEmpty(self):
        return self.size == 0

    def isFull(self):
        return self.size == len(self.buffer)

    def get(self):
        if self.isEmpty():
            return None
        item = self.buffer[self.getPointer.getAndIncrement()]
        self.size -= 1
        return item

    def put(self, item):
        if item is None:
            raise ValueError("Null items are not allowed")
        if self.isFull():
            self.getPointer.getAndIncrement()
        else:
            self.size += 1
        self.buffer[self.putPointer.getAndIncrement()] = item

    class CircularPointer:
        def __init__(self, pointer, max):
            self.pointer = pointer
            self.max = max

        def getAndIncrement(self):
            tmp = self.pointer
            self.pointer = (self.pointer + 1) % self.max
            return tmp

if __name__ == "__main__":
    import sys
    scanner = sys.stdin
    bufferSize = int(scanner.readline())
    numberOfOperations = int(scanner.readline())
    buffer = Main(bufferSize)
    for i in range(numberOfOperations):
        operation = scanner.readline().strip()
        if operation == "PUT":
            element = scanner.readline().strip()
            buffer.put(element)
            print("Added: " + element)
        elif operation == "GET":
            result = buffer.get()
            print("Retrieved: " + (result if result is not None else "null"))
        elif operation == "ISEMPTY":
            print("Is Empty: " + str(buffer.isEmpty()))
        elif operation == "ISFULL":
            print("Is Full: " + str(buffer.isFull()))

